### PR TITLE
Directly send logged out Suclass users to TPA when trying to enroll

### PR DESCRIPTION
**Pull request by kluo**
Default flow of sending to register page is unwanted for Suclass as
we don't want users trying to create non-SUNet accounts on it

@stvstnfrd @caesar2164

### CODE REVIEW

**Comment by stvstnfrd** [comment]
I'm confused. When we discussed this yesterday, the consensus was that the behavior on stage was what we wanted.
Why are we pursuing a different approach now?

**Comment by kluo** [comment]
This case still exists. The agreement we came to yesterday was for just going to the home page, while this is going directly to a course about page while logged out.

File: lms/templates/courseware/course_about.html [hunk 1]
```diff
@@ -109,7 +109,11 @@
           location.href = xhr.responseText;
         }
       } else if (xhr.status == 403) {
-          location.href = "${reverse('register_user')}?course_id=${course.id | u}&enrollment_action=enroll";
+          %if settings.SHIB_ONLY_SITE:
+            location.href = "/login?next=${reverse('finish_auth') + '?course_id=' + course.id.to_deprecated_string() + '&enrollment_action=enroll&next=' + reverse('dashboard') | u}";
```
Comment (caesar2164) [inline lms/templates/courseware/course_about.html]: Do we need the `course.id.to_deprecated_string`?

Comment (kluo) [reply lms/templates/courseware/course_about.html]: Yes otherwise it errors because it can't convert a CourseLocator type object to unicode

Comment (caesar2164) [reply lms/templates/courseware/course_about.html]: huh, but then why does the original not use `course.id.to_deprecated_string`?

Comment (kluo) [reply lms/templates/courseware/course_about.html]: When there is no string concatenation it gets cast normally, the issue only arises with concatenation

Comment (caesar2164) [reply lms/templates/courseware/course_about.html]: Gotcha

**Comment by caesar2164** [comment]
👍

**Comment by stvstnfrd** [comment]
Please don't merge this until after we have discussed.

**Comment by stvstnfrd** [comment]
Deprecating in favor of #729

**Pull request closed without merging**